File: _shared/typescript/TreeNode.ts
export class TreeNode {
  val: number
  left: TreeNode | null
  right: TreeNode | null
  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
  }
}

export function inOrder(head: TreeNode) {
  const vals: number[] = [];

  (function inner(head: TreeNode | null) {
    if (head === null) {
      return;
    }

    inner(head.left);
    vals.push(head.val);
    inner(head.right);
  })(head)

  return vals;
}

File: _shared/tree_node.py
from typing import List, Optional


class TreeNode:

    def __init__(self,
                 val: int = 0,
                 left: Optional['TreeNode'] = None,
                 right: Optional['TreeNode'] = None):
        self.val: int = val
        self.left: Optional['TreeNode'] = left
        self.right: Optional['TreeNode'] = right


def inorder_traversal(root: Optional[TreeNode]) -> List[int]:
    memo = []

    def inner(_root: Optional[TreeNode]):
        if _root is None:
            return

        inner(_root.left)
        memo.append(_root.val)
        inner(_root.right)

    inner(root)
    return memo
